compute/instances.go

```diff
@@ -314,8 +320,33 @@ func (input *CreateInstanceInput) toAPI() (map[string]interface{}, error) {
 		result["image"] = input.Image
 	}
 
+	// If we are passed []string from input.Networks that do not conflict with networks provided by NetworkObjects,  add them to the request sent to CloudAPI
+	var networks []NetworkObject
+
+	if len(input.NetworkObjects) > 0 {
+		networks = append(networks, input.NetworkObjects...)
+	}
+
 	if len(input.Networks) > 0 {
-		result["networks"] = input.Networks
+		for _, v := range input.Networks {
+			if len(networks) > 0 {
+				for _, n := range networks {
+					if n.IPv4UUID != v {
```

Comment:
I think this logic is not quite right.  As it is, it seems like you're looping over all of the `NetworkObject` entries in the `networks` array, and comparing the UUID string `v` (from `input.Networks`) against the UUID for that entry.  If the UUID doesn't match, you're then appending a network object for that UUID.

I believe, though, that the if the _first_ network doesn't match (but a _subsequent_ one would) you're going to add the network to the list in error.  Also, if there are, say, three networks in the list that don't match `v`, you're going to add this network object three times?

I think we could make this whole thing a bit simpler, including skipping the `len(networks) > 0` check on L332, and the `len(input.Networks) > 0` check on L330, by going with something more like:

```
for _, netuuid := range input.Networks {
    found := false

    for _, net := range networks {
        if net.IPv4UUID == netuuid {
            found = true
        }
    }

    if !found {
        networks = append(networks, NetworkObject{
            IPv4UUID: netuuid,
        })
    }
}
```

I believe `for ... range` won't actually do anything unless there are entries in `input.Networks` and/or `networks` already, so the length checks aren't needed.  Tracking whether we've found an existing entry in `found`, then only acting on it at the end, means we'll not add a duplicate by accident.  Does that all make sense?